packages/context/src/context.ts

```diff
@@ -248,17 +306,22 @@ export class Context {
    *
    * @param keyWithPath The binding key, optionally suffixed with a path to the
    *   (deeply) nested property to retrieve.
-   * @param session An object to keep states of the resolution
+   * @param options Options for resolution or a session
    * @returns The bound value or a promise of the bound value, depending
    *   on how the binding was configured.
    * @internal
    */
   getValueOrPromise(
     keyWithPath: string,
-    session?: ResolutionSession,
+    options?: ResolutionOptions | ResolutionSession,
```

Comment:
I find this line confusing: `options` may be set to a `ResolutionSession`?

I am proposing to rename the argument to `optionsOrSession`, update tsdoc to mention the backwards-compatibility API, and then change the code below to the following:

```ts
const options: ResolutionOptions = optionsOrSession instanceof ResolutionSession 
  ? {session: optionsOrSession} : optionsOrSession;
```

Reply:
I'm wondering if it's worth to pile up backward-compatible tricks at this stage. Is there a better way to deal with it, for example, start to use semantic versioning for stable modules, such as `metadata` and `context`?

Reply:
> I'm wondering if it's worth to pile up backward-compatible tricks at this stage.

I think we should not worry about backwards compatibility at this stage at all. Just make sure the commit message contains `BREAKING CHANGE` footer explaining what no longer works and how to upgrade - this information will be added by our release tooling to release notes.

>  Is there a better way to deal with it, for example, start to use semantic versioning for stable modules, such as metadata and context?

Let's discuss here: https://github.com/strongloop/loopback-next/issues/954